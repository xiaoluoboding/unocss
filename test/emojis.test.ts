import { createGenerator } from '@unocss/core'
import presetAttributify from '@unocss/preset-attributify'
import { describe, expect, test } from 'vitest'

describe('emojis', () => {
  const fixture1 = `
    <button 
    游불 class="游불" 游불="1" 游볳-2 type="button"
    >
    Button
    </button>
  `
  const uno = createGenerator({
    presets: [
      presetAttributify({ strict: true }),
    ],
    rules: [
      ['游불', {
        color: 'red',
      }],
    ],
  })

  test('extractor1', async() => {
    expect(await uno.applyExtractors(fixture1)).toMatchSnapshot()
  })

  test('fixture1', async() => {
    const { css } = await uno.generate(fixture1)
    expect(css).toMatchSnapshot()
  })
})
