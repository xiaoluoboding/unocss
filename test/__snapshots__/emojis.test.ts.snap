// Vitest Snapshot v1

exports[`emojis > extractor1 1`] = `
Set {
  "[游불=\\"\\"]",
  "游불",
  "[游불~=\\"1\\"]",
  "[游볳-2=\\"\\"]",
  "[type~=\\"button\\"]",
}
`;

exports[`emojis > fixture1 1`] = `
"/* layer: default */
.游불,[游불=\\"\\"]{color:red;}"
`;
